flask_login/login_manager.py

```diff
@@ -326,7 +326,7 @@ def _load_user(self):
                 return self._load_from_cookie(request.cookies[cookie_name])
             elif self.request_callback:
                 return self._load_from_request(request)
-            elif header_name in request.headers:
+            elif self.header_callback and header_name in request.headers:
```

Comment:
I don't like moving the check for `self.header_callback` here since it seems to belong near the `_load_from_header` code.